lms/static/js/groups/views/cohorts.js

```diff
@@ -45,13 +53,17 @@
             }));
         },
 
-        onSync: function() {
+        onSync: function(model, response, options) {
             var selectedCohort = this.lastSelectedCohortId && this.model.get(this.lastSelectedCohortId),
                 hasCohorts = this.model.length > 0,
                 cohortNavElement = this.$('.cohort-management-nav'),
-                additionalCohortControlElement = this.$('.wrapper-cohort-supplemental');
+                additionalCohortControlElement = this.$('.wrapper-cohort-supplemental'),
+                isModelUpdate = options && options.patch && response.hasOwnProperty('user_partition_id');
```

Comment:
You could possibly override the sync method of the cohort model (or collection) to set a property indicating whether it was a model or collection sync.